src/jit/emitxarch.cpp

```diff
@@ -9929,7 +9929,7 @@ BYTE* emitter::emitOutputRI(BYTE* dst, instrDesc* id)
                 break;
 
             case IF_RRW_CNS:
-                assert(id->idGCref() == GCT_BYREF);
+                assert((id->idGCref() == GCT_BYREF) || (id->idGCref() == GCT_GCREF));
```

Comment:
What is the reason (or test case) for allow a GCREF here?

This is an instruction that is doing a Read/Write/Modify operation using a constant right?
Such an operation wouldn't typically produce a GCREF.